vagrant/Vagrantfile

```
@@ -30,111 +31,194 @@ unless errors.empty?
+          ansible_host: '192.168.50.21',
+          ansible_ssh_private_key_file: "/home/vagrant/.ssh/node1.key"
+        },
+        node2: {
+          openshift_ip: '192.168.50.22',
+          openshift_node_labels: "\"{'region': 'primary', 'zone': 'west'}\"",
+          openshift_schedulable: true,
+          ansible_host: '192.168.50.22',
+          ansible_ssh_private_key_file: "/home/vagrant/.ssh/node2.key"
+        },
+        admin1: {
+          ansible_connection: 'local'
+        }
+      }
     end
   end
 
-  if deployment_type == 'enterprise'
-    config.vm.provision "shell", inline: "echo \"setting up Red Hat subscriptions\""
-    config.vm.provision "shell", path: "repos.sh"
+
+  if deployment_type == 'openshift-enterprise'
+    subscription_steps = <<-SUB_STEPS
+      echo "Setting up Red Hat subscriptions"
+      POOL_ID=$(sudo subscription-manager list --available | sed -n '/Employee SKU/,/System Type/p' | grep "Pool ID" | tail -1 | cut -d':' -f2 | xargs)
+      echo -e $GREEN"Trying PoolID: $POOL_ID"$WHITE
+      sudo subscription-manager attach --pool=$POOL_ID
+      sudo subscription-manager repos --disable="*"
+      sudo subscription-manager repos --enable="rhel-7-server-rpms" --enable="rhel-7-server-extras-rpms" --enable="rhel-7-server-ose-3.3-rpms"
+    SUB_STEPS
+  else
+    subscription_steps = ''
   end
-  config.vm.provision "shell", inline: "echo \"running install_helper_packages.sh\""
-  config.vm.provision "shell", path: "install_helper_packages.sh"
-  config.vm.provision "shell", inline: "echo \"running fix_hosts_file.sh\""
-  config.vm.provision "shell", path: "fix_hosts_file.sh"
+
+  config.vm.provision "shell", inline: <<-SHELL
+    #{subscription_steps}
+
+    sudo yum -y install git bash-completion
+
+    # Vagrant's "change host name" capability for Fedora
+    # maps hostname to loopback, conflicting with hostmanager.
+    # We must repair /etc/hosts
+    sudo sed -i '/127\.0\.0\.1\s.*example.*/d' /etc/hosts
```

@detiber Certainly for me sed and vagrant do not play nicely for me and I only get the sed to work if I put it in an external script file.

Not sure how important fixing the /etc/hosts file is. I do know that the playbook will complain when run if /etc/hosts still contains the loopback mapping to the hostname but that you can ignore it. Since ansible local is not running interactively it automatically ignores the error.

Can you check whether the /etc/hosts file is being fixed in your setup? There should be no

127.0.0.1  master1.example.com master1

line when you logon to master1 (etc.)

See

https://github.com/mitchellh/vagrant/issues/5822

---

Ah, I was thinking about converting those tasks to Ansible anyway, will try to get that cleaned up shortly.